### DISCUSSION TITLE
If Kõnele is launched through AnySoftKeyboard dictation, it opens a web search instead of inputting text

### DISCUSSION THREAD

**Discussion by marmistrz** [Kaljurand/K6nele#110] [Q&A]
Kõnele (using WhisperInput from [this fork](https://github.com/gkfabs/WhisperInput/tree/mine/main)) is set as my speech-to-text provider in the system settings. I installed AnySoftKeyboard from F-Droid.

If I tap the mic icon in the AOSP keyboard, it launches Konele speak & swipe and works as expected.
If I tap the mic icon in AnySoftKeyboard, it launches the Konele popup "Say what to search" and a web search is executed instead of inputting the text.

**Comment by Kaljurand** [comment]
I think it's rather a question to AnySoftKeyboard: what is the exact semantics of its mic icon (under various conditions, e.g. Google's recognizer is not present). ASK has a lot of (old) related open issues, so the situation seems to be unclear and needs to be documented first by ASK:

- https://github.com/AnySoftKeyboard/AnySoftKeyboard/issues/3230
- https://github.com/AnySoftKeyboard/AnySoftKeyboard/issues/1672
- https://github.com/AnySoftKeyboard/AnySoftKeyboard/issues/716
- https://github.com/AnySoftKeyboard/AnySoftKeyboard/issues/927
- https://github.com/AnySoftKeyboard/AnySoftKeyboard/issues/2686

However, if the Kõnele pop-up performs a web search, instead of returning the transcription to the caller, then this might be an issue in Kõnele, but this should first be demonstrated with other apps. I wasn't able to observe it now, e.g. in GMail, SwiftKey, etc.

**Comment by Redsandro** [comment]
> using WhisperInput from [this fork](https://github.com/gkfabs/WhisperInput/tree/mine/main)

Interesting. What is the functional difference with original WhisperInput?

**Comment by Kaljurand** [comment]
> > using WhisperInput from [this fork](https://github.com/gkfabs/WhisperInput/tree/mine/main)
> 
> Interesting. What is the functional difference with original WhisperInput?

As far as I can see (from the source code comparison https://github.com/alex-vt/WhisperInput/compare/main...gkfabs:WhisperInput:mine/main) the fork offers multilinguality. But it's better to ask @gkfabs, and move the discussion to a different thread.
